pallets/subtensor/src/lib.rs

```diff
@@ -1305,6 +1307,81 @@ pub mod pallet {
             Self::do_set_weights(origin, netuid, dests, weights, version_key)
         }
 
+        // # Args:
+        // 	* `origin`: (<T as frame_system::Config>Origin):
+        // 		- The caller, a hotkey who wishes to set their weights.
+        //
+        // 	* `netuid` (u16):
+        // 		- The network uid we are setting these weights on.
+        //
+        // 	* `hotkey` (T::AccountId):
+        // 		- The hotkey associated with the operation and the calling coldkey.
+        //
+        // 	* `dests` (Vec<u16>):
+        // 		- The edge endpoint for the weight, i.e. j for w_ij.
+        //
+        // 	* 'weights' (Vec<u16>):
+        // 		- The u16 integer encoded weights. Interpreted as rational
+        // 		values in the range [0,1]. They must sum to in32::MAX.
+        //
+        // 	* 'version_key' ( u64 ):
+        // 		- The network version key to check if the validator is up to date.
+        //
+        // # Event:
+        //
+        // 	* WeightsSet;
+        // 		- On successfully setting the weights on chain.
+        //
+        // # Raises:
+        //
+        // 	* NonAssociatedColdKey;
+        // 		- Attempting to set weights on a non-associated cold key.
+        //
+        // 	* 'NetworkDoesNotExist':
+        // 		- Attempting to set weights on a non-existent network.
+        //
+        // 	* 'NotRootSubnet':
+        // 		- Attempting to set weights on a subnet that is not the root network.
+        //
+        // 	* 'WeightVecNotEqualSize':
+        // 		- Attempting to set weights with uids not of same length.
+        //
+        // 	* 'InvalidUid':
+        // 		- Attempting to set weights with invalid uids.
+        //
+        // 	* 'NotRegistered':
+        // 		- Attempting to set weights from a non registered account.
+        //
+        // 	* 'NotSettingEnoughWeights':
+        // 		- Attempting to set weights with fewer weights than min.
+        //
+        //  * 'IncorrectNetworkVersionKey':
+        //      - Attempting to set weights with the incorrect network version key.
+        //
+        //  * 'SettingWeightsTooFast':
+        //      - Attempting to set weights too fast.
+        //
+        // 	* 'NotSettingEnoughWeights':
+        // 		- Attempting to set weights with fewer weights than min.
+        //
+        // 	* 'MaxWeightExceeded':
+        // 		- Attempting to set weights with max value exceeding limit.
+        //
+        #[pallet::call_index(8)]
```

Comment:
I know we haven't really started doing this yet but ideally these should be doc comments